### DISCUSSION TITLE
Not populated fields should not be returned in the result

### DISCUSSION THREAD

**Discussion by gijsmin** [mikro-orm/mikro-orm#4642] [Q&A]
**Describe the bug**
I'm currently trying to find an entry and populate some relation fields, but not every field in the relation should be returned, but that's exactly what happens. 

I've created an endpoint in my NestJS application where I give an `id` as a parameter and it should only return specific fields to the result.

**To Reproduce**
```
return this.clientRepository.findOne(
	{ idx: index },
	{
		fields: [
			"id",
			"firstName",
			"lastName",
			"email",
			"phone",
			"createdBy.id",
			"createdBy.firstName",
			"createdBy.lastName",
		],
	},
)
```

**Expected behavior**
```
{
    "firstName": "Vincenza",
    "lastName": "Langosh",
    "email": "Garfield_Mraz19@yahoo.com",
    "phone": "1-677-326-5688",
    "createdBy": {
        "id": 1,
        "firstName": "Test",
        "lastName": "User"
    }
}
```

**Actual behavior**
```
{
    "firstName": "Vincenza",
    "lastName": "Langosh",
    "email": "Garfield_Mraz19@yahoo.com",
    "phone": "1-677-326-5688",
    "createdBy": {
        "id": 1,
        "createdAt": null,
        "firstName": "Test",
        "lastName": "User",
        "email": null,
        "avatar": null,
        "lastLogin": null,
        "deletedAt": null,
        "verifiedAt": null
    }
}
```
All the fields from the `createdBy` should not be included in the result except the ones in the `fields` array.

**Comment by B4nan** [comment]
Implicit serialization is already reworked in v6 via #4203, and this should no longer reproduce there. If it does, please provide a complete reproduction, this is too sparse.
